biodatasets/quaero/quaero.py

```diff
@@ -0,0 +1,280 @@
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _get_bioc_entity(self, span, doc_text, db_id_key=None):
+        """Parse BioC entity annotation.
+        Parameters
+        ----------
+        span : BioCAnnotation
+            BioC entity annotation
+        doc_text : string
+            document text, required to construct text spans
+        db_id_key : str, optional
+            database name used for normalization, by default "MESH"
+        Returns
+        -------
+        dict
+            entity information
+        """
+        offsets = [(loc.offset, loc.offset + loc.length) for loc in span.locations]
+        texts = [doc_text[i:j] for i, j in offsets]
+        db_ids = span.infons[db_id_key] if db_id_key else "-1"
+        normalized = [
+            # some entities are linked to multiple normalized ids
+            {"db_name": db_id_key, "db_id": db_id}
+            for db_id in db_ids.split("|")
+        ]
+
+        return {
+            "id": span.id,
+            "offsets": offsets,
+            "text": texts,
+            "type": span.infons["type"],
+            "normalized": normalized,
+        }
+
+
+    def _get_document_text(self, xdoc):
+        """Build document text for unit testing entity span offsets."""
+        text = ""
+        for passage in xdoc.passages:
+            pad = passage.offset - len(text)
+            text += (" " * pad) + passage.text
+        return text
+
+    def _split_generators(self, dl_manager):
+        urls = _URLS[self.config.name]
+        data_dir = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "test"
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    def _generate_examples(self, filepath, split):
+        key=0
+        if self.config.schema == "source":
+            for file in [f"EMEA_{split}_bioc", f"MEDLINE_{split}_bioc"]:
+              f = os.path.join(filepath, "QUAERO_BioC", "corpus", split, file)
+              reader = bioc.BioCXMLDocumentReader(str(f))
```

Comment:
Always because of the `bioc` update, this should be changed to `bioc.biocxml.BioCXMLDocumentReader`